### PR TITLE
Feat reload projects page on project creation

### PR DESCRIPTION
**Pull request by Mubangizi**
#### What this PR does
- Reload projects page on project creation
- Add spinner to project create button
- Add feed-back for Project creation

### CODE REVIEW

File: src/components/UserProjectsPage/index.jsx [hunk 1]
```diff
@@ -43,17 +43,12 @@ class UserProjectsPage extends React.Component {
...
-    const { error } = this.state;
-    this.setState({
-      [e.target.name]: e.target.value
-    });
-
-    if (error) {
-      this.setState({
-        error: ''
-      });
+    if (isAdded !== prevProps.isAdded) {
```
Comment (MutegekiHenry) [inline src/components/UserProjectsPage/index.jsx:49]: @Mubangizi This component reloads the page so fast that the user doesn't get a chance to see the success creation message, so i was requesting you add a one-second delay to cater for the success message display. somethig like
`  setTimeout(
      () => {
        if (isAdded !== prevProps.isAdded) {
          getUserProjects(data.id);
          this.hideForm();
        }
      }, 1000
    );
  }
`

Comment (StephenAraka) [reply src/components/UserProjectsPage/index.jsx:49]: @MutegekiHenry. I think it's okay without the success message.
If you remember from the conversation we had on one of the calls, we said setTimeout could introduce bugs. Therefore, it is better we get rid of them. In this case, the component re-rendering with the new project is sufficient indication of success.

Comment (Mubangizi) [reply src/components/UserProjectsPage/index.jsx:49]: yeah me too, i find it more convenient when you see the component, load and go off, telling you that you request was successful

File: src/redux/reducers/addProjectReducer.js [hunk 2]
```diff
@@ -1,36 +1,58 @@
...
       ...state,
       isFailed: true,
       isAdded: false,
-      errorOccured: action.payload.error,
+      isAdding: false,
+      errorCode: action.payload.errorCode,
       message: 'Failed to add Project'
     };
 
+  case CLEAR_ADD_PROJECT_STATE:
```
[the remark this answers is not in the document]
Comment (Mubangizi) [reply src/redux/reducers/addProjectReducer.js:46]: No it does actually clear them, am not sure what you are talking about, but it changes them back. I had to test it before i added it

[the remark this answers is not in the document]
Comment (MutegekiHenry) [reply src/redux/reducers/addProjectReducer.js]: maybe its just my pc cache

**Pull request merged**